pkg/restclient/device.go

```diff
@@ -1,17 +1,34 @@
 package restclient
 
 import (
+	"fmt"
 	"net/http"
+	"strings"
 
 	deviceapi "github.com/gluster/glusterd2/plugins/device/api"
 )
 
 // DeviceAdd registers devices
-func (c *Client) DeviceAdd(peerid string, device string) (deviceapi.AddDeviceResp, error) {
+func (c *Client) DeviceAdd(peerid, device string) (deviceapi.AddDeviceResp, error) {
 	var peerinfo deviceapi.AddDeviceResp
 	req := deviceapi.AddDeviceReq{
 		Device: device,
 	}
 	err := c.post("/v1/devices/"+peerid, req, http.StatusOK, &peerinfo)
 	return peerinfo, err
 }
+
+// DeviceDelete removes devices
+func (c *Client) DeviceDelete(peerid, device string) error {
+	device = strings.TrimLeft(device, "/")
+	url := fmt.Sprintf("/v1/devices/%s/%s", peerid, device)
+	return c.del(url, nil, http.StatusNoContent, nil)
+}
+
+// DeviceList lists the devices
+func (c *Client) DeviceList(peerid string) ([]deviceapi.Info, error) {
```

Comment:
This is repeated in PR #1118

Reply:
Yes I required this for the e2e test cases, will rebase once that PR is merged.